clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1976,7 +2000,7 @@ void Sema::MarkDevice(void) {
         case attr::Kind::SYCLIntelNoGlobalWorkOffset:
         case attr::Kind::SYCLSimd: {
           if ((A->getKind() == attr::Kind::SYCLSimd) &&
-              !KernelBody->getAttr<SYCLSimdAttr>()) {
+              KernelBody && !KernelBody->getAttr<SYCLSimdAttr>()) {
```

Comment:
Do we need an assert here that the KernelBody can't be NULL?  A previous case statement has a check for it.

Reply:
I'm not sure this is strictly enforced. Let's leave a check for now.